rbx_binary/src/deserializer.rs

```diff
@@ -1123,6 +1120,86 @@ impl<R: Read> BinaryDeserializer<R> {
                     })
                 }
             },
+            Type::OptionalCFrame => match canonical_type {
+                VariantType::OptionalCFrame => {
+                    let referents = &type_info.referents;
+                    let mut rotations = Vec::with_capacity(referents.len());
+
+                    // Roblox writes a type marker for CFrame here that we don't
+                    // need to use.
+                    chunk.read_u8()?;
```

Comment:
Let's assert that this value is what we expect. It's weird that it exists, but I'm a little worried that this type ID might just mean "Optional <something>" and we're intended to ascribe meaning to this byte. It could be used in the future to have more optional value types.